bundles/org.openhab.binding.revogismartstripcontrol/src/main/java/org/openhab/binding/revogismartstripcontrol/internal/RevogiSmartStripControlHandler.java

```diff
@@ -0,0 +1,168 @@
+            case PLUG_4_SWITCH:
+                switchPlug(command, 4);
+                break;
+            case PLUG_5_SWITCH:
+                switchPlug(command, 5);
+                break;
+            case PLUG_6_SWITCH:
+                switchPlug(command, 6);
+                break;
+            case ALL_PLUGS:
+                switchPlug(command, 0);
+                break;
+        }
+    }
+
+    private void switchPlug(Command command, int port) {
+        if (config == null) {
+            logger.warn("No config available, config object was null");
+            return;
+        }
+        if (command instanceof OnOffType) {
+            int state = convertOnOffTypeToState(command);
+            switchService.switchPort(config.getSerialNumber(), config.ipAddress, port, state);
+        }
+    }
+
+    private int convertOnOffTypeToState(Command command) {
+        if (command == OnOffType.ON) {
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(RevogiSmartStripControlConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Example for background initialization:
+        scheduler.execute(this::updateStripInformation);
+        Runnable runnable = RevogiSmartStripControlHandler.this::updateStripInformation;
+
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            pollingJob = scheduler.scheduleWithFixedDelay(runnable, 0, config.getPollInterval(), TimeUnit.SECONDS);
+        }
+
+        logger.debug("Finished initializing!");
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+    }
+
+    private void updateStripInformation() {
+        if (config == null) {
+            logger.warn("No config available, config object was null");
+            return;
+        }
+        Status status = statusService.queryStatus(config.getSerialNumber(), config.getIpAddress());
+        if (status.isOnline()) {
+            updateStatus(ThingStatus.ONLINE);
+            handleAllPlugsInformation(status);
+            handleSinglePlugInformation(status);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.GONE,
+                    "Retrieved status code: " + status.getResponseCode());
+        }
+    }
+
+    private void handleSinglePlugInformation(Status status) {
+        for (int i = 0; i < status.getSwitchValue().size(); i++) {
+            int plugNumber = i + 1;
+            updateState("plug" + plugNumber + "#switch", OnOffType.from(status.getSwitchValue().get(i).toString()));
+            updateState("plug" + plugNumber + "#watt", new DecimalType(status.getWatt().get(i) / 1000f));
+            updateState("plug" + plugNumber + "#amp", new DecimalType(status.getAmp().get(i) / 1000f));
```

Comment:
These are quantity type channels:
```suggestion
            updateState("plug" + plugNumber + "#watt", new QuantityType(status.getWatt().get(i), MILLI(WATT)));
            updateState("plug" + plugNumber + "#amp", new QuantityType(status.getAmp().get(i), MILLI(AMPERE)));
```